sw/device/lib/testing/test_framework/coverage_llvm.c

```diff
@@ -37,6 +37,7 @@ void coverage_send_buffer(void) {
   if (buf_size_u64 > sizeof(buf)) {
     LOG_ERROR("ERROR: LLVM profile buffer is too large: %u bytes.",
               (uint32_t)buf_size_u64);
+    OT_UNREACHABLE();
```

Comment:
Isn't this reserved for truly impossible branches? The documentation states that behavior is undefined if control reaches this point — I'm not sure if this means the entire branch, including LOG_ERROR is now undefined.

Reply:
Good point, and you're technically right but here the compiler generates an `unimp` which triggers an exception. This was basically a quick fix to make the currently running test fail if we fail to collect coverage due to infrastructure instead of silently ignoring it. I think the right solution would be update the prototype to return a `bool` and handle that in the `main()`. I'll come back to this after finishing up other tasks. Captured in #18748